### PR TITLE
Feature/ID-288/User account crud operations 'Create'

### PR DESCRIPTION
**Pull request by JeferssonCL**
Author
Samuel Escalera @SamuelEscalera
Jefersson Coronel @JeferssonCL
## What?
We did:

connection to the database, to the collection users
UI to view account information
All registered users will be automatically added to the database.
Option for the user to edit his: fullName and userName(backend and UI implementation only)
Option to delete the currently logged in account (backend and UI implementation only)
##Why?
To have a record of users in the database

## CodeReview
@DiegoFigueroaSevillano
@fabianromeroclaros
@Karvlox
Taiga: https://tree.taiga.io/project/marcel-morales-hermes/us/288

## Trainer
@mmoralesv

### CODE REVIEW

**Review by DiegoFigueroaSevillano — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/controller/Utiils/ImageUtil.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        int size = Math.min(width, height);
+
+        int x = (width - size) / 2;
+        int y = (height - size) / 2;
+
+        return Bitmap.createBitmap(bitmap, x, y, size, size);
+    }
+
+    public static Uri getUriFromBitmap(Context context, Bitmap bitmap) {
```
Comment (fabianromeroclaros) [inline app/src/main/java/com/isc/hermes/controller/Utiils/ImageUtil.java:31]: Missing javadocs

File: app/src/main/java/com/isc/hermes/database/ApiHandler.java [hunk 2]
```diff
@@ -34,6 +36,14 @@ public Future<String> getFutureCollectionString(String params) {
         );
     }
 
+    public Future<?> postFutureCollections(String params, Object object) {
```
Comment (fabianromeroclaros) [inline app/src/main/java/com/isc/hermes/database/ApiHandler.java:39]: Missing javadocs

Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/database/ApiHandler.java:46]: Resolved

File: app/src/main/java/com/isc/hermes/controller/Utiils/ImageUtil.java (hunk 1 above)
Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/controller/Utiils/ImageUtil.java:38]: Resolved

File: app/src/main/java/com/isc/hermes/AccountInformation.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+     */
+    private void generateActionToComboBox() {
+        generateComponentsToComboBox().setOnItemClickListener((parent, view, position, id) -> {
+            String item = parent.getItemAtPosition(position).toString();
+            Toast.makeText(getApplicationContext(), "Item: " + item,
+                    Toast.LENGTH_SHORT).show();
+        });
+    }
+
+    private void updateTextFieldsByUser() {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/AccountInformation.java:70]: Javadocs missing

File: app/src/main/java/com/isc/hermes/AccountInformation.java [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+     * @param view The view that triggers the navigation.
+     */
+    public void editAccountAction(View view) {
+        buttonSaveInformation.setVisibility(View.VISIBLE);
+        textFieldFullName.setEnabled(true);
+        comboBoxField.setEnabled(true);
+        textFieldUserName.setEnabled(true);
+    }
+
+    private  void updateInformationUser() {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/AccountInformation.java:180]: Javadocs missing

File: app/src/main/java/com/isc/hermes/MainActivity.java [hunk 5]
```diff
@@ -287,6 +304,15 @@ protected void onSaveInstanceState(@NonNull Bundle outState) {
...
+    /**
+     * Retrieves the user information passed through the intent.
+     * Gets the Parcelable "userObtained" extra from the intent and assigns it to the userRegistered variable.
+     */
+    private void getUserInformation() {
+        Intent intent = getIntent();
+        userRegistered = intent.getParcelableExtra("userObtained");
+    }
+
     public void openStylesMenu(View view) {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/MainActivity.java:316]: Javadocs missing

File: app/src/main/java/com/isc/hermes/controller/PopUp/PopUp.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    public PopUp(Activity activity, TypePopUp typePopUp){
+        super(activity);
+        this.activity = activity;
+        initializeDialog();
+        initializeRefuseButton();
+        initializeConfirmButton();
+        assignValuesToComponents(typePopUp);
+    }
+
+    public void assignValuesToComponents(TypePopUp typePopUp) {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/controller/PopUp/PopUp.java:40]: Javadocs missing

**Review by fabianromeroclaros — APPROVED** [review]

File: app/src/main/java/com/isc/hermes/model/User.java [hunk 7]
```diff
@@ -44,6 +44,13 @@ protected User(Parcel in) {
         id = in.readString();
     }
 
+    public User(String email, String fullName, String userName, String typeUser) {
```
Comment (Karvlox) [inline app/src/main/java/com/isc/hermes/model/User.java:47]: Javadocs missing

Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/model/User.java:55]: Resolved

File: app/src/main/java/com/isc/hermes/controller/PopUp/PopUp.java (hunk 6 above)
Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/controller/PopUp/PopUp.java:45]: Resolved

File: app/src/main/java/com/isc/hermes/MainActivity.java (hunk 5 above)
Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/MainActivity.java:321]: Resolved

File: app/src/main/java/com/isc/hermes/AccountInformation.java (hunk 4 above)
Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/AccountInformation.java:186]: Resolved

File: app/src/main/java/com/isc/hermes/AccountInformation.java (hunk 3 above)
Comment (SamuelEscalera) [reply app/src/main/java/com/isc/hermes/AccountInformation.java:73]: Resolved

**Review by Karvlox — APPROVED** [review]

**Pull request merged**